pdns/dnsdistdist/dnsdist-backend.cc

```diff
@@ -1052,6 +1052,13 @@ void ServerPool::addServer(shared_ptr<DownstreamState>& server)
     serv.first = idx++;
   }
   *servers = std::make_shared<const ServerPolicy::NumberedServerVector>(std::move(newServers));
+
+  if ((*servers)->size() == 1) {
+    d_tcpOnly = server->isTCPOnly();
+  }
+  else if (d_tcpOnly && !server->isTCPOnly()) {
```

Comment:
Lefthand expression is redundant.
```suggestion
  else if (!server->isTCPOnly()) {
```